### PR TITLE
[nfc] Fix RTTI for `InstrProf` intrinsics

### PR DESCRIPTION
**Pull request by mtrofin**

### CODE REVIEW

File: llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp
Comment (ellishg) [inline llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp:1]: It looks like changes from a different diff got mixed in here

Comment (mtrofin) [reply llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp:1]: ugh. it's what happens if I write code too late in the day :)

**Comment by mtrofin** [comment]
@ellishg - is this good to go? thanks!

File: llvm/include/llvm/IR/IntrinsicInst.h [hunk 1]
```diff
@@ -1429,7 +1429,35 @@ class VACopyInst : public IntrinsicInst {
...
+    }
+    return false;
+  }
+
 public:
+  static bool classof(const Value *V) {
+    if (const auto *Instr = dyn_cast<IntrinsicInst>(V))
+      return isCounterBase(*Instr) || isMCDCBitmapBase(*Instr) ||
+             Instr->getIntrinsicID() ==
+                 Intrinsic::instrprof_mcdc_condbitmap_update;
```
Comment (ellishg) [inline llvm/include/llvm/IR/IntrinsicInst.h:1458]: I wonder if this should be in `isMCDCBitmapBase()` CC @evodius96

Comment (ellishg) [reply llvm/include/llvm/IR/IntrinsicInst.h:1458]: Nevermind, I found the comment for `InstrProfMCDCCondBitmapUpdate`

**Review by ellishg — APPROVED** [review]
LGTM

**Pull request merged**